presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java

```diff
@@ -51,6 +60,7 @@ public ColumnMetadata(String name, Type type, String comment, String extraInfo,
         this.comment = comment;
         this.extraInfo = extraInfo;
         this.hidden = hidden;
+        this.properties = (properties != null && !properties.isEmpty()) ? Collections.unmodifiableMap(new LinkedHashMap<>(properties)) : Collections.emptyMap();
```

Comment:
The Presto code base generally rejects nulls rather than being null-friendly, so we should not allow null here. Users should instead pass an empty collection. For example, `ConnectorTableMetadata` does not accept null properties.

Nit: static import `unmodifiableMap` and `emptyMap` for readability, as the method names are self-descriptive without the class prefix